File: 0061-rotate-list/0061-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if( k == 0 || head == NULL || head->next == NULL) return head;
        int len=1;
        ListNode *cur = head;
        while(cur->next != NULL){
            len++;
            cur=cur->next;
        }
        k = len - (k%len);
        cur->next = head;
        while(k > 0){
            cur=cur->next;
            k--;
        } 
        head = cur->next;
        cur->next = NULL;
        return head;
    }
};

File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if( k == 0 || head ==null || head.next == null) return head;
        int len=1;
        ListNode cur=head;
        while(cur.next != null){
            len++;
            cur=cur.next;
        }
        k = len - (k%len);
        cur.next = head;
        while(k > 0){
            cur=cur.next;
            k--;
        } 
        head = cur.next;
        cur.next = null;
        return head;
    }
}